staging/src/k8s.io/kubectl/pkg/cmd/debug/debug.go

```diff
@@ -403,6 +417,10 @@ func (o *DebugOptions) debugByEphemeralContainer(ctx context.Context, pod *corev
 	klog.V(2).Infof("new ephemeral container: %#v", debugContainer)
 	debugPod := pod.DeepCopy()
 	debugPod.Spec.EphemeralContainers = append(debugPod.Spec.EphemeralContainers, *debugContainer)
+	if err := o.profile.Apply(debugPod, debugContainer.Name, pod); err != nil {
```

Comment:
Shouldn't that move over to `generateDebugContainer`? Alternatively each profile should be applied right after `generate...` invocation. Currently there are inconsistencies.

Reply:
Updated